src/pages-and-resources/discussions/app-config-form/AppConfigForm.jsx

```diff
@@ -1,51 +1,58 @@
 import React, {
-  useCallback, useContext, useEffect, useRef,
+  useCallback, useContext, useEffect,
 } from 'react';
 import PropTypes from 'prop-types';
 import { useDispatch, useSelector } from 'react-redux';
 import { useRouteMatch } from 'react-router';
-import { Container } from '@edx/paragon';
 import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
 import { history } from '@edx/frontend-platform';
+import { Container } from '@edx/paragon';
+
 import { useModel } from '../../../generic/model-store';
-import { fetchAppConfig, saveAppConfig } from './data/thunks';
-import LegacyConfigForm from './apps/legacy';
-import LtiConfigForm from './apps/lti';
-import messages from './messages';
 import { PagesAndResourcesContext } from '../../PagesAndResourcesProvider';
-import AppConfigFormProvider from './AppConfigFormProvider';
+import { LOADED, selectApp } from '../data/slice';
+import { saveAppConfig } from '../data/thunks';
+
+import messages from './messages';
+import AppConfigFormProvider, { AppConfigFormContext } from './AppConfigFormProvider';
 import AppConfigFormApplyButton from './AppConfigFormApplyButton';
+import LegacyConfigForm from './apps/legacy';
+import LtiConfigForm from './apps/lti';
 
 function AppConfigForm({
   courseId, intl,
 }) {
-  const { params: { appId: routeAppId } } = useRouteMatch();
-  const formRef = useRef();
+  const dispatch = useDispatch();
+  const { formRef } = useContext(AppConfigFormContext);
```

Comment:
This was a straight-up bug.  We should have been using the ref from AppConfigFormContext all along.